go-server-server/go/models.go

```diff
@@ -211,7 +215,7 @@ func (m *RouteModel) UnmarshalJSON(data []byte) (err error) {
         return
     }
 
-    if required.MACAddress != nil {
+    if required.IfName == nil && required.MACAddress != nil {
```

Comment:
Is this really required?

Reply:
We need this check because MACAddress should be validated only when IfName is not provided, otherwise it might throw an error

---
In reply to: [506645497](https://github.com/Azure/sonic-restapi/pull/51#discussion_r506645497) [](ancestors = 506645497)